### PR TITLE
Added Junit Tests for classes in service/api module and Entity, Event, Locator package

### PR DESCRIPTION
**Pull request by sonja-ct**
Added Junit tests for KapuaIllegalArgumentException class
Added Junit tests for KapuaRuntimeException class
Added Junit Tests for Entity package
Added Junit Tests for Event package
Added Junit Tests for Locator package

Signed-off-by: Sonja <sonja.matic@comtrade.com>

**Related Issue**
/

**Description of the solution adopted**
/

**Screenshots**
/

**Any side note on the changes made**
/

### CODE REVIEW

File: service/api/src/test/java/org/eclipse/kapua/KapuaIllegalArgumentExceptionTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                        "Error: " + name + ", " + value, "Max number of " + name + " reached. Please increase the number or set InfiniteChild" + name + " parameter to True.", "The start time cannot be later than the end time.",
+                        "Start and end time cannot be at the same point in time. Please change the parameters.", "There should only be one scheduling parameter - either Retry interval or Cron schedule.",
+                        "The selected devices were not found. Please refresh device list.", "An entity with the same value for field already exists here or in another job.",
+                        "Error: Another resource is currently downloading. Please wait a few moments so package finishes downloading and then try again.",
+                        "Error: " + name + ", " + value, "Operation not allowed on admin role.", "Operation not allowed on this specific permission.", "The Service is disabled: " + name};
+
+                for (int i = 0; i < kapuaErrorCodes.length; i++) {
```
Comment (LeoNerdoG) [inline service/api/src/test/java/org/eclipse/kapua/KapuaIllegalArgumentExceptionTest.java:72]: Please use enhanced for loop: "                for (KapuaErrorCodes ker: kapuaErrorCodes) {"

File: service/api/src/test/java/org/eclipse/kapua/KapuaRuntimeExceptionTest.java [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+    @Test
+    public void internalErrorCauseTest() {
+        String message = "Message";
+        Throwable[] throwables = {new Throwable(message), new Throwable()};
+        Throwable nullThrowable = null;
+        String[] expectedMessage = {"An internal error occurred: " + message + ".", "An internal error occurred: " + throwables[1] + "."};
+        String[] arguments = {"Message", "java.lang.Throwable"};
+
+        for (int i = 0; i < throwables.length; i++) {
```
Comment (LeoNerdoG) [inline service/api/src/test/java/org/eclipse/kapua/KapuaRuntimeExceptionTest.java:184]: Please use enhanced for loop: for(Throwable thr:throwables)

File: service/api/src/test/java/org/eclipse/kapua/KapuaRuntimeExceptionTest.java [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    @Before
+    public void initialize() {
+        expectedErrorMessage = "kapua-service-error-messages";
+        kapuaErrorCode = KapuaErrorCodes.ADMIN_ROLE_DELETED_ERROR;
+        argument1 = "user";
+        argument2 = 1;
+        argument3 = 'c';
+        throwables = new Throwable[]{new Throwable(), null};
+    }
+
```
Comment (LeoNerdoG) [inline service/api/src/test/java/org/eclipse/kapua/KapuaRuntimeExceptionTest.java:38]: Please delete empty line.

File: service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        expectedMessageNullArguments = new String[]{"Service unavailable {0}", "{0} is not a valid service provider for {1}", "Factory unavailable {0}",
+                "{0} is not a valid factory provider for {1}", "Error: ", "Error: ", "Error: "};
+        expectedMessage = new String[]{"Service unavailable " + argument1, argument1 + " is not a valid service provider for " + argument2, "Factory unavailable " + argument1,
+                argument1 + " is not a valid factory provider for " + argument2, "Error: " + argument1 + ", " + argument2 + ", " + argument3, "Error: " + argument1 + ", " + argument2 + ", " + argument3,
+                "Error: " + argument1 + ", " + argument2 + ", " + argument3};
+    }
+
+    @Test
+    public void kapuaLocatorExceptionKapuaLocatorErrorCodesParameterTest() {
+        for (int i = 0; i < kapuaLocatorErrorCodes.length; i++) {
```
Comment (LeoNerdoG) [inline service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java:49]: Please use enhanced for loop.

File: service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        KapuaLocatorException kapuaLocatorException = new KapuaLocatorException(null);
+        assertNull("Null expected.", kapuaLocatorException.getCode());
+        assertNull("Null expected.", kapuaLocatorException.getCause());
+        assertEquals("Expected and actual values should be the same.", expectedErrorMessage, kapuaLocatorException.getKapuaErrorMessagesBundle());
+        kapuaLocatorException.getMessage();
+    }
+
+    @Test
+    public void kapuaLocatorExceptionKapuaLocatorErrorCodesObjectParametersTest() {
+        for (int i = 0; i < kapuaLocatorErrorCodes.length; i++) {
```
Comment (LeoNerdoG) [inline service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java:69]: Please use enhanced for loop.

File: service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+            assertEquals("Expected and actual values should be the same.", kapuaLocatorErrorCodes[i], kapuaLocatorException.getCode());
+            assertNull("Null expected.", kapuaLocatorException.getCause());
+            assertEquals("Expected and actual values should be the same.", expectedMessage[i], kapuaLocatorException.getMessage());
+            assertEquals("Expected and actual values should be the same.", expectedErrorMessage, kapuaLocatorException.getKapuaErrorMessagesBundle());
+        }
+    }
+
+    @Test
+    public void kapuaLocatorExceptionKapuaLocatorErrorCodesNullObjectParametersTest() {
+        for (int i = 0; i < kapuaLocatorErrorCodes.length; i++) {
```
Comment (LeoNerdoG) [inline service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java:80]: Please use enhanced for loop.

File: service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+        assertNull("Null expected.", kapuaLocatorException.getCode());
+        assertNull("Null expected.", kapuaLocatorException.getCause());
+        assertEquals("Expected and actual values should be the same.", expectedErrorMessage, kapuaLocatorException.getKapuaErrorMessagesBundle());
+        kapuaLocatorException.getMessage();
+    }
+
+    @Test
+    public void kapuaLocatorExceptionKapuaLocatorErrorCodesThrowableObjectParametersTest() {
+        for (Throwable throwable : throwables) {
+            for (int i = 0; i < kapuaLocatorErrorCodes.length; i++) {
```
Comment (LeoNerdoG) [inline service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java:110]: Please use enhanced for loop.

File: service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+                assertEquals("Expected and actual values should be the same.", expectedMessage[i], kapuaLocatorException.getMessage());
+                assertEquals("Expected and actual values should be the same.", expectedErrorMessage, kapuaLocatorException.getKapuaErrorMessagesBundle());
+            }
+        }
+    }
+
+    @Test
+    public void kapuaLocatorExceptionKapuaLocatorErrorCodesThrowableNullObjectParametersTest() {
+        for (Throwable throwable : throwables) {
+            for (int i = 0; i < kapuaLocatorErrorCodes.length; i++) {
```
Comment (LeoNerdoG) [inline service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java:123]: Please use enhanced for loop.

**Review by LeoNerdoG — CHANGES_REQUESTED** [review]
hey @sonja-ct,
I have reviewed this branch, and there are couple of minor things to be fixed, please take a look.

File: service/api/src/test/java/org/eclipse/kapua/KapuaIllegalArgumentExceptionTest.java (hunk 1 above)
Comment (sonja-ct) [reply service/api/src/test/java/org/eclipse/kapua/KapuaIllegalArgumentExceptionTest.java:72]: I used index i to get values from different arrays in the same position

File: service/api/src/test/java/org/eclipse/kapua/KapuaRuntimeExceptionTest.java (hunk 2 above)
Comment (sonja-ct) [reply service/api/src/test/java/org/eclipse/kapua/KapuaRuntimeExceptionTest.java:184]: I used index i to get values from different arrays in the same position

File: service/api/src/test/java/org/eclipse/kapua/KapuaRuntimeExceptionTest.java (hunk 3 above)
Comment (sonja-ct) [reply service/api/src/test/java/org/eclipse/kapua/KapuaRuntimeExceptionTest.java:38]: I used index i to get values from different arrays in the same position

File: service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java (hunk 5 above)
Comment (sonja-ct) [reply service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java:69]: I used index i to get values from different arrays in the same position

File: service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java (hunk 6 above)
Comment (sonja-ct) [reply service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java:80]: I used index i to get values from different arrays in the same position

File: service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java (hunk 4 above)
Comment (sonja-ct) [reply service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java:49]: I used index i to get values from different arrays in the same position

File: service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java (hunk 7 above)
Comment (sonja-ct) [reply service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java:110]: I used index i to get values from different arrays in the same position

File: service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java (hunk 8 above)
Comment (sonja-ct) [reply service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java:123]: I used index i to get values from different arrays in the same position

**Comment by LeoNerdoG** [comment]
Hey @Coduz,
I have reviewed this PR and it looks OK. Travis is green and coverage increased, can you please take a look at this and merge when you can? 
Thanks!

**Review by LeoNerdoG — APPROVED** [review]

**Review by Coduz — APPROVED** [review]

**Pull request merged**